src/room_membership/join_room/v1.rs

```diff
@@ -0,0 +1,88 @@
+//! [POST /_synapse/admin/v1/join/:room_id_or_alias](https://github.com/element-hq/synapse/blob/master/docs/admin_api/room_membership.md)
+
+use ruma::{
+    api::{request, response, Metadata},
+    metadata, OwnedRoomId, OwnedRoomOrAliasId, OwnedUserId,
+};
+use serde::{Deserialize, Serialize};
+
+const METADATA: Metadata = metadata! {
+    method: POST,
+    rate_limited: false,
+    authentication: AccessToken,
+    history: {
+        unstable => "/_synapse/admin/v1/join/:room_id_or_alias",
+    }
+};
+
+#[request]
+#[derive(Serialize, Deserialize, PartialEq)]
+pub struct Request {
+    /// Alias or ID of the room to join.
+    #[ruma_api(path)]
+    pub room_id_or_alias: OwnedRoomOrAliasId,
+
+    /// User to join the room.
+    pub user_id: OwnedUserId,
+}
+
+#[response]
+#[derive(Serialize, Deserialize, PartialEq)]
+pub struct Response {
+    /// Room ID of the joined room.
+    pub room_id: OwnedRoomId,
+}
+
+impl Request {
+    /// Creates a new `Request` with the given room or alias ID and user id.
+    pub fn new(room_id_or_alias: OwnedRoomOrAliasId, user_id: OwnedUserId) -> Self {
+        Self { room_id_or_alias, user_id }
+    }
+}
+
+impl Response {
+    /// Creates a new `Response` with the given room id
+    pub fn new(room_id: OwnedRoomId) -> Self {
+        Self { room_id }
+    }
+}
+
+#[test]
+fn test_join_room() {
+    use ruma::{RoomId, UserId};
+    use std::convert::TryFrom;
+
+    let room_id: &RoomId =
+        <&RoomId>::try_from("!test:example.com").expect("Failed to create RoomId");
+
+    let user_id: &UserId =
+        <&UserId>::try_from("@carl:example.com").expect("Failed to create UserId.");
```

Comment:
The latest version of ruma (that I just merged an upgrade for) contains `owned_room_id!` and `owned_user_id!` macros that you can use here.

Reply:
Awesome thanks